proposals/dht-js.md

```diff
@@ -0,0 +1,106 @@
+# Effective content and service discovery in JS
+
+Authors: @jacobheun
+
+Initial PR: TBD
+
+## Purpose &amp; impact
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives.
+-->
+
+**Current State**
+- JS projects rely on delegate and preload nodes to be able to interact with the live network
+- PL hosted delegate/preload nodes are often overloaded negatively impact performance of projects
```

Comment:
The preload nodes also garbage collect the content periodically so over time the content becomes inaccessible

Reply:
Sorry if I'm just totally missing the idea here, but why are preload nodes and garbage collection on them relevant in this discussion? Aren't they only really needed if a node is unreachable (e.g. in a browser)?

Reply:
The point above is that the PL hosted infra is often overloaded which is a risk to project functionality.

I was making the additional point that even if the nodes were not overloaded, they garbage collect anything uploaded to them, so the IPFS magic of 'add a file to your node, cat it from another node' is time-limited if the only way that content makes it to another node is via a preload node.

Reply:
I'm confused, this doc states.

> The focus of this effort is to build the DHT to function solely in Node.js. Running a DHT in browser is not currently viable.

Putting data in preload nodes is not required to get NodeJS to play well with the rest of the network.